### PR TITLE
SSHOperator ignores cmd_timeout (#27182)

### LINKED ISSUE #27182: SSHOperator ignores cmd_timeout
### Apache Airflow Provider(s)

ssh

### Versions of Apache Airflow Providers

_No response_

### Apache Airflow version

2.4.1

### Operating System

linux

### Deployment

Other

### Deployment details

_No response_

### What happened

Hi,

SSHOperator documentation states that we should be using cmd_timeout instead of timeout
```
:param timeout: (deprecated) timeout (in seconds) for executing the command. The default is 10 seconds.
        Use conn_timeout and cmd_timeout parameters instead.
```

But the code doesn't use cmd_timeout at all - and it's still passing `self.timeout` when running the ssh command:
```
return self.ssh_hook.exec_ssh_client_command(
            ssh_client, command, timeout=self.timeout, environment=self.environment, get_pty=self.get_pty
        )
```

It seems to me that we should `self.cmd_timeout` here instead. When creating the hook, it correctly uses `self.conn_timeout`.

I'll try to work on a PR for this.

### What you think should happen instead

_No response_

### How to reproduce

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by punx120**
closes: #27182 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/ssh/hooks/ssh.py
Comment (o-nikolas) [inline airflow/providers/ssh/hooks/ssh.py:499]: I think you need to check if **all** three results are empty to be sure a timeout has occurred. Useful link: https://stackoverflow.com/a/15195460/1055702

Comment (punx120) [reply airflow/providers/ssh/hooks/ssh.py:499]: I'm not sure - from `select` doc
```
the return value is a tuple of three lists corresponding to the first three arguments; 
each contains the subset of the corresponding file descriptors that are ready.
```
and we pass empty list to 2nd and 3rd args.

Comment (o-nikolas) [reply airflow/providers/ssh/hooks/ssh.py:499]: I don't have a deep enough understanding of the select api and ssh to know for sure, but it seems harmless to check the others.

If you're positive that this is correct then I'm happy to commit and approve.

Comment (punx120) [reply airflow/providers/ssh/hooks/ssh.py:499]: Since we're passing empty list for the 2nd and 3rd arg, i dont see how the other 2 cannot be empty. It makes sense to pass empty file descriptors since the 2nd arg is for writing and the 3rd for exception condition.

Comment (o-nikolas) [reply airflow/providers/ssh/hooks/ssh.py:499]: Okay, sounds good, approving :)

**Review by o-nikolas — APPROVED** [review]

File: airflow/providers/ssh/hooks/ssh.py
Comment (punx120) [reply airflow/providers/ssh/hooks/ssh.py:499]: what's the next step to get this merged?

Comment (o-nikolas) [reply airflow/providers/ssh/hooks/ssh.py:499]: > what's the next step to get this merged?

There are some static checks that are failing which need fixing before it can be merged. You can enable pre-commit to run static checks on your code before submitting a PR update ([readme](https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst#installing-pre-commit-hooks)) 

Then someone with committer status will need to review/approve/merge the change :) (CC: @eladkal @potiuk)

Comment (punx120) [reply airflow/providers/ssh/hooks/ssh.py:499]: got it - i think i fixed it - re-running now

**Review by potiuk — APPROVED** [review]

File: airflow/providers/ssh/hooks/ssh.py
Comment (punx120) [reply airflow/providers/ssh/hooks/ssh.py:499]: Thank you @potiuk anything else we need to merge it?

**Comment by eladkal** [comment]
I think we can just remove timeout parameter. 

https://github.com/apache/airflow/blob/64174ce25ae800a38e712aa0bd62a5893ea2ff99/airflow/providers/ssh/operators/ssh.py#L102-L109

The parameter is deprecated. Next release is a breaking change one so isn't it better to remove this parameter and adjust the fix accordingly?

**Comment by punx120** [comment]
> I think we can just remove timeout parameter.
> 
> https://github.com/apache/airflow/blob/64174ce25ae800a38e712aa0bd62a5893ea2ff99/airflow/providers/ssh/operators/ssh.py#L102-L109
> 
> The parameter is deprecated. Next release is a breaking change one so isn't it better to remove this parameter and adjust the fix accordingly?

Sure i can look into that

**Comment by punx120** [comment]
> > I think we can just remove timeout parameter.
> > https://github.com/apache/airflow/blob/64174ce25ae800a38e712aa0bd62a5893ea2ff99/airflow/providers/ssh/operators/ssh.py#L102-L109
> > 
> > The parameter is deprecated. Next release is a breaking change one so isn't it better to remove this parameter and adjust the fix accordingly?
> 
> Sure i can look into that

Done!

**Comment by punx120** [comment]
@eladkal @potiuk Can one of you please approve so we can merge it?

**Pull request merged**
